cmd/crank/beta/top/top.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2023 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package top contains the top command.
+package top
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/alecthomas/kong"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/metrics/pkg/client/clientset/versioned"
+
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+)
+
+// TODO:(piotr1215) implement kubeconfig using ctrl "sigs.k8s.io/controller-runtime"
+func getKubeConfig() (string, error) {
+	kubeconfig := os.Getenv("KUBECONFIG")
+	if kubeconfig != "" {
+		return kubeconfig, nil
+	}
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return "", fmt.Errorf("could not get user home directory: %w", err)
+	}
+	return filepath.Join(home, ".kube", "config"), nil
+}
+
+// Cmd represents the top command.
+type Cmd struct{}
+
+// Help returns help instructions for the top command.
+func (c *Cmd) Help() string {
+	return `Usage: crossplane top`
+}
+
+// Run runs the top command.
+func (c *Cmd) Run(_ *kong.Context, logger logging.Logger) error { //nolint:gocyclo // TODO:(piotr1215) refactor to lower cyclomatic complexity
+	// TODO:(piotr1215) add logging in key places
+	logger = logger.WithValues("cmd", "top")
+
+	// Build the config from the kubeconfig path
+	kubeconfig, err := getKubeConfig()
```

Comment:
you can use the existing helper function instead see: https://github.com/crossplane/crossplane/blob/ff75e37237ca2f3085c2217e8dc45caaccd57db3/cmd/crank/beta/trace/trace.go#L109

Reply:
Thank you for commenting @phisco , I have left a TODO comment in code to do just that and more refactoring, this is very much WIP :).